### DISCUSSION TITLE
Ideas, balance suggestions, and more

### DISCUSSION THREAD

**Discussion by TGWeaver** [TGWeaver/CDDA-Sky-Islands#14] [Ideas]
Given the lack of permanent alternatives, this will serve as a central place to share your non-bug-related feedback about the mod, at least for now.

Any ideas you have for new features, suggestions about balance or difficulty, complaints about game flow, or even just comments to make about the existing state of the mod -- anything like that can go here.

**Comment by Marc13Bautista** [comment]
Any plan to perhaps add a LMOE shelter start scenario to the character start for Sky Islands? 
*Maybe labeled as "Sky Island (LMOE)" or "Flying Fortress"

In this sense, it would work like vanilla CDDA's LMOE sheltered/safe place start, except the island would have the topside LMOE hill bunker entrance at the (top?) center of the island (with an unlocked door) and using the ladder, it would bring you underground to one of the predetermined base layouts/variations available for the LMOE shelter'(and selectable at the last tab for character creations).

I think this would be a great way for beginning players like myself who would like to skip over the early stages of base building from scratch and get right to having a home that's not disturbed by the heavy winds/winter/etc.

Though, given that with this objectively easier start, perhaps the functions of the LMOE shelter could be that the (heavy) equipment (ex: Treadmill?) inside would be (near) broken and gives incentive to go out and repair those things or something. 

Also, portal storms would be a threat again as maybe you could disable the portal storm item that makes them go away on the island as you have safety underground. Maybe not. 

I've no idea if the sky island it big enough to accomodate the underground layer of an LMOE shelter (considering z levels and how deep the island goes), but it would be really awesome to have a floating bunker home start. 

Thanks for this awesome mod. Been having a blast playing with no hope mod.

**Comment by TGWeaver** [comment]
Would just a simple underground shelter work, rather than an entire LMOE as it currently exists? I could do something like this but would ideally make it without most furniture those locations start with, and probably a smaller space in total, with fewer rooms or potentially just one.
I had initially considered an underground base instead of an island base, but the lack of solar or wind power seemed a turnoff, so this could get benefits from both.

**Reply by Marc13Bautista** [reply]
That can work. Though having starter furniture and allowing players to expand and move things to their personal preference is a good alternative as you suggest. 

Having the best of both worlds with farming and solar on the surface with a shelter underground was what I envisioned with a LMOE start. Maybe if you go through with the underground shelter, leave what you think are the essentials. Maybe even a generator (gasoline powered?) underground but the player would need to find more gas themselves.

Another thought I had was making keycarded rooms in the underground shelter that allow players to get starter/mid/end game goodies. 

Such as finding a military keycard will open up an armory for a generous amount of weapons/attachments, and so on for the other keycards that can be looted in the world. 

Just some thoughts. 👍

**Comment by ziroc** [comment]
Some feedback. I have a great time with this mod :)
But I am wondering how accessible is the late game. Bionics probably are no-go as you cannot easily get to Rubik and make him trust you. This leaves mutation. I find various labs during the raids, but with the limited amount of time, I don't think I can clear them and make way to the good stuff.

**Reply by TGWeaver** [reply]
If you can find an autodoc, you can basically steal it and put it on your island, if you use a late-game warp item, the copyplate. Grab an autoclave as well and it should be possible to install bionics at home.
As for labs, they're definitely a huge time investment, but every 2 Earthbound pills you take are equivalent to an entire raid's worth of extra time. You could potentially eat a bunch and then have enough time to loot a lab, but it's a very high investment. Sort of an ultimate goal scenario I suppose.

**Reply by ziroc** [reply]
The autodoc solution is fine, but as you probably know, most bionics are now gated behind the exodii and Rubik. There are some bionics in the labs, but mostly low tier I think, and not much choice.
And you need quite a lot of time  - to talk with Rubik every other day to build up the relationship. Which is not possible here.
Similar problem with the mutagen components. If you want high tier mutations you need to raid quite a lot of labs to gather enough components. 

I want to say again, I enjoy the mod very much, it is your decision if you want to make these late game power-ups available or not. I just think that currently, they are beyond reach.

**Reply by Marc13Bautista** [reply]
Hoping this may be resolved with a static npc trader (akin to the "intercom" npcs from one of the factions) where you can can grind out shards from raids to trade in for end game loot that is at a very large price. It is near impossible to raise reputation and such with the mechanics of limited time we can spend with factions aside from a quick chat... Though I'm unsure if this is present in the game but with a phone, maybe it was possible to call faction leaders or traders you had already visited for the first time at least once and are put on the "explored" factions list. Need someone to confirm that. 

TGWeavee, maybe something could be done about this whether it would be a npc trader intercom that stays on sky island (perhaps in the bunker (locked behind some keycard?) with all available factions' loot pools offered at extremely high prices of shards, or having the access to some npc trader teleport behind a mini trading post that will be swapped out with the factions that were explored by the player on a daily/weekly basis or even direct teleportation manually for some shards. 

I'm not familiar with how the game is coded but these are just my thoughts for this situation. Hoping the underground shelter start could be implemented, and with the a intercom trader it would be awesome.

**Reply by TGWeaver** [reply]
My understanding is that the intercom is a semi-hardcoded thing that only reflects a nearby human NPC anyway. That said, no, I actually wasn't aware bionics had been so heavily revamped.

What about adding a random bionic treasure as a craftable reward? In the same way that you can currently craft a random artifact, you'd craft a sort of lootbox that you could unpack for a random bionic, maybe within a specific category. Due to the random nature and the possibility of duplicates, the overall price per "roll" would be relatively low. Maybe to balance that out, the recipe would be locked behind a sort of mission or goal that has to be completed? A tall barrier to entry but then the individual price isn't so bad.
Are mutagens still craftable? Could do a similar thing with those even.

**Reply by Marc13Bautista** [reply]
Sounds like a good way to get them, though I've never experienced endgame with bionics myself. I'm not so sure about mutagens either. 

But I still think a seamless way to progress that balances out Vanilla CDDA and this mod without much conflict could be in the additional underground shelter below the island with keycard locked rooms where greater variety of missions can be unlocked after unlocking such rooms with specific key cards that can be found on raids or custom keycards that can be mission set and made from this mod. 

In key carded rooms, you can find some starter gear that pertains to their theme and perhaps even some sort of lottery loot box machine that outputs equipment or items using shards that are obtained. If that's possible. 

Such as a military keycard room where a few random pistols or melee weapons can be found lying around, and a military lottery machine that outputs special weapons at a very low rate up to rng. A medical/waste treatment plant keycard room can have starter IFAKs and a machine for bionics. 

I don't know what other key cards there are available in CDDA but it would be a cool concept that can be perfected on. 

Maybe using your suggestion of a crafting recipe for specific categorized loot boxes that is only obtained behind their specific keycard rooms would work better though.

**Reply by TGWeaver** [reply]
I'm not sure how many of these ideas could even work, code aside. "A machine for bionics" is a very vague way of handwaving that bionics should be accessible, somehow. "Greater variety of missions" is easy to say but very hard to implement, especially since most mission formats are either to kill a target or to retrieve an item, the latter of which would require custom handling for every objective, and wouldn't add much more than crafting recipes do already.

But frankly, even implementation aside, that whole idea is just way too much. For one, I'm trying to avoid giving the player any gear they can find themselves -- weapons, gear, meds, etc., so any rewards are more about enabling you to bring those things home after you already find them. For another, a system like you describe would take _many_ times longer to design and code than the entire mod has taken so far. And most importantly of all, I'm now trying to keep the mod fairly vanilla in the sense of having it rely on the base game's content and mechanics as much as possible.

**Reply by Marc13Bautista** [reply]
That's fair enough. I guess the mod is just quite a lot more difficult with the lowered chances to find items when using the No Hope mod (but a lot more fun). I didn't even realize until later after a new save that No Hope would affect the spawn rates at the start for the free tent and this mod's stone items.

Yeah, thinking back on it after a while it would be a bit too much to make this sort of underground shelter with free goodies and adding more to an already good mission system. I guess I'm just inexperienced with base building from scratch and should pursue making an awesome log house using the inifinitree you added. It's already good with the Vanilla game (and no hope mod) as it currently is. 

Though I'm still curious to how you will tackle NPCs.

**Reply by ziroc** [reply]
I feel like the intercom or a random NPC on the island don't get along with the theme of the mod.

After some more playing, I have an idea about the mutagen.
What about a token that can be bought and the effect is - the next time you raid, you are sent near a lab (or other scientific location) and given more time - ex. 24 hours. You generally need to raid quite a few labs to get enough resources to craft high end mutagens. Also you need 4 (heavy) tools from the labs. So this solution will not be too over powered. Also labs can be hard, so its a big risk to do this mission without some better armor and weapons and skills.
I am not sure how to balance the time, some labs are multi-level and have many enemies.
Also I don't know if it is possible to prevent you from going outside the science lab location, so you cannot use this mission for regular looting.

I don't have an idea how to handle the bionics stuff though.

**Reply by Onkl** [reply]
Since there is already an earthbound pill you are able to extend you stay on the world below. Of course you'll have a problem in a huge central lab; you won't be able to finish it. But this is the point of this whole mod, isn't it? To be under a strict time constraint.

I wouldn't change anything about it as it is now in the game. I've played maybe 30h with Sky Island mod enabled and love it the way it is. 

There are possibilities to get CBMs without going to labs... 

You can get an autoclave from the top level of any trans coast logistics along with the autoclave pouches.
You can get CBMs from Bio Operators (Found in military outposts) 
You can get CBMs from Zomborgs (They drop the best ones)
The Exodii

Maybe I'll change my mind when I think it gets boring and I don't get a chance to get to the real highend stuff. But so far, this mod is a blast. It should be in the base game!

**Comment by Sajuukkar** [comment]
Questions and suggestion:

1: Are the earthbound pills _supposed_ to rot? I'm not complaining per se but my character is VERY displeased with my unwillingness to throw them out >.>

2: I'm not sure if this intended or not but currently the warped hauler harness can contain container items, including garbage bags and body bags.... I mean, I know that SOUNDS reasonable at first glance but once it dawned on me I could put things IN them while they're in the harness I realized it became very broken very quickly, so hey, if intended, GREAT! I love the idea of hauling back 100L of foodstuffs with almost no weight penalty XD but I highly suspect it's not intended so yeah, heads up.

As for the suggestion: could a furniture copyplate be introduced as a thing? not that it's very useful game mechanics wise but i'd like the idea of installing bathtubs, showers, toilets to my sky home as well as the other, non-buildable stuff. Ideally, I'd like to do these things without debug like we currently have to for the autodoc couch.

**Comment by LHOAGE** [comment]
Just here to get a discussion going. Nothing here stated is concrete. Mostly tossing ideas around to see if it fits, is needed or if there's a different direction. These ideas are coming from someone who has limited time to play and is not keen on rerolling or resetting constantly. These may be better solutions that are built in with the difficulty adjustments at the statue rather than default settings.

### Healing

Wound healing could have some functionality added lore wise that can take care of a soft lock situation due to RNG of expeditions.  As it stands now, it is entirely possible to make it unrealistic or impossible to recover from multiple deaths post 10 expeditions. I have a possible solution that is geared more towards novice players or victims of unfortunate RNG.

### Primary Solution

Create a conditional where the player arrives to the return statue naked and holding nothing. The avatar takes pity on the player and admires their idealism instead of their materialism. The statue grants full health as the statue services does. The 50 material tokens are confiscated and the player is immediately returned back home so they cannot game the system and sneak in look post recovery. 

### Issues To Consider:

Player can just reroll death over and over until a portal is right next to spawn. Defeating the difficulty.  

### Secondary Solutions

Consider temporary mental abnormalities that occur if the player dies too many times too soon. 

Consider a conditional to check the distance the portal is from the character. If it is beyond X overmap tiles, allow reward. If it is not, the player does not receive avatar blessing.

Consider teleporting the player to an ant farm or something similar that requires close distance for aggro to give the player a fighting chance of fleeing, if the player departs after a recent death, is post 10 expeditions, is post death health rating, has no unarmed combat and/or dodge rating of 5 or more,

### Food and Hunger

Unless I am unaware, I can't find any device to prevent "practical" thirst and hunger soft lock due to the previous issue's mentioned conditions that occur from healing softlock. Hunger and thirst carry over from death to death. This can create a situation where the player is soft locked out of the game due to the detrimental effects of thirst and hunger.

### Primary Solution

Create a conditional where the character departs naked and holding nothing and returns to the statue naked after having drank salt water from a marsh and has a pinecone in their hand. The player must be thirsty or worse and underweight. The avatar takes pity on the player and admires their yearn to suffer for the greater good. The player has their thirst and hunger reset to maximum but their weight is not affected. The 50 material shards are confiscated and the player is immediately returned back to home.

### Issues To Consider

See Healing

### Secondary Solutions

See Healing.
Player cannot perform both rituals in one expedition. They must choose if both system conditions are met.

### OR

Ignore all that and make a difficulty option to allow the statue to replenish both health and thirst/hunger. Or some variation.

**Comment by Stricknine47** [comment]
Is it possible to play this mod with the Last Generation fork?

**Comment by 5batek5** [comment]
Suggestions:
Blacklist all quest locations (if not already).
Add a trader entity on island to sell items for something interesting.
Mission to find random rare item.
